src/shogun/features/DenseFeatures.cpp

```diff
@@ -265,23 +267,51 @@ template<class ST> void CDenseFeatures<ST>::feature_subset(int32_t* idx, int32_t
 	}
 }
 
-template<class ST> SGMatrix<ST> CDenseFeatures<ST>::get_feature_matrix()
+template <class ST>
+SGMatrix<ST> CDenseFeatures<ST>::get_feature_matrix()
 {
 	if (!m_subset_stack->has_subsets())
 		return feature_matrix;
 
-	SGMatrix<ST> submatrix(num_features, get_num_vectors());
+	return deep_copy_feature_matrix(SGMatrix<ST>(), 0);
```

Comment:
Haven't changed the behavior. It returns the same underlying matrix when there are no subsets. Please check the above return statement that I haven't touched.